modules/collect/src/main/java/com/opengamma/strata/collect/result/ValueWithFailures.java

```diff
@@ -0,0 +1,398 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.result;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * A value with associated failures.
+ * <p>
+ * This captures a common use case where an operation can tolerate some failure.
+ * This is often referred to as partial success or partial failure.
+ * The class stores the value, of any object type, and a list of failures that may be empty.
+ * <p>
+ * The success result must be able to handle the case where everything fails.
+ * In most cases, the success result will be a collection type, such as {@link List}
+ * or {@link Map}, which can be empty if operation failed completely.
+ * <p>
+ * The classic example is loading rows from a file, when some rows are valid and some are invalid.
+ * The partial result would contain the successful rows, with the list of failures containing an
+ * entry for each row that failed to parse.
+ *
+ * @param <T> the type of the underlying success result, typically a collection type
+ */
+@BeanDefinition(builderScope = "private")
+public final class ValueWithFailures<T>
+    implements ImmutableBean, Serializable {
+
+  /**
+   * The success value.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final T value;
```

Comment:
This is referred to interchangeably as "result" and "value" - should probably pick "value" given the name of the class and this field.